util/importers/fisher.py

```diff
@@ -0,0 +1,300 @@
+    
+    # Conditionally split Fisher data into train/validation/test sets
+    _maybe_split_sets(data_dir, "fisher-2004-wav", "fisher-2004-wav-splits")
+    _maybe_split_sets(data_dir, "fisher-2005-wav", "fisher-2005-wav-splits")
+
+    # Create train DataSet
+    train = _read_data_set(graph, data_dir, "fisher-200?-wav/train", thread_count, batch_size, numcep, numcontext)
+
+    # Create dev DataSet
+    dev = _read_data_set(graph, data_dir, "fisher-200?-wav/dev", thread_count, batch_size, numcep, numcontext)
+
+    # Create test DataSet
+    test = _read_data_set(graph, data_dir, "fisher-200?-wav/test", thread_count, batch_size, numcep, numcontext)
+
+    # Return DataSets
+    return DataSets(train, dev, test)
+
+def _maybe_convert_wav(data_dir, original_data, converted_data):
+    source_dir = os.path.join(data_dir, original_data)
+    target_dir = os.path.join(data_dir, converted_data)
+    
+    # Conditionally convert sph files to wav files
+    if not os.path.exists(target_dir):
+        # Create target_dir
+        os.makedirs(target_dir)
+        
+        # Loop over sph files in source_dir and convert each to 16-bit PCM wav
+        for root, dirnames, filenames in os.walk(source_dir):
+            for filename in fnmatch.filter(filenames, "*.sph"):
+                sph_file = os.path.join(root, filename)
+                wav_filename = os.path.splitext(os.path.basename(sph_file))[0] + ".wav"
+                wav_file = os.path.join(target_dir, wav_filename)
+                print("converting {} to {}".format(sph_file, wav_file))
+                subprocess.check_call(["sph2pipe", "-p", "-f", "rif", sph_file, wav_file])
+
+def _parse_transcriptions(trans_file):
+    segments = []
+    with open(trans_file, "r") as fin:
+        for line in fin:
+            if line.startswith("#") or len(line) <= 1:
+                continue
+            
+            start_time_beg = 0
+            start_time_end = line.find(" ", start_time_beg)
+            
+            stop_time_beg = start_time_end + 1
+            stop_time_end = line.find(" ", stop_time_beg)
+            
+            speaker_beg = stop_time_end + 1
+            speaker_end = line.find(" ", speaker_beg)
+            
+            transcript_beg = speaker_end + 1
+            transcript_end = len(line)
+            
+            segments.append({
+                "start_time": float(line[start_time_beg:start_time_end]),
+                "stop_time": float(line[stop_time_beg:stop_time_end]),
+                "speaker": line[speaker_beg:speaker_end],
+                "transcript": line[transcript_beg:transcript_end].strip(),
+            })
+    return segments
+
+def _maybe_split_wav(data_dir, original_data, converted_data):
+    source_dir = os.path.join(data_dir, original_data)
+    target_dir = os.path.join(data_dir, converted_data)
+    
+    if os.path.exists(target_dir):
```

Comment:
Is this check correct?

For example, it seem like the previous call
```python
_maybe_convert_wav(data_dir, "LDC2004S13", "fisher-2004-wav")
```
will create the directory, line 139, this call
```python
_maybe_split_wav(data_dir, os.path.join("LDC2004T19", "fe_03_p1_tran", "data", "trans"), "fisher-2004-wav")
```
is looking for. Thus is seems like the code below
```python
    # Loop over transcription files and split corresponding wav
    for root, dirnames, filenames in os.walk(source_dir):
      ...
```
will never run.